### ISSUE TITLE
E/linker: library "/system/lib/libopencv_core.so" ("/system/lib/libopencv_core.so") needed or .....

### ISSUE THREAD

**Issue by 805447391** [bytedeco/javacv#906]
i try to take JavaCV library to my Android application in two ways:
1.put all related .jar to libs
2.put wrapper jar to libs and put .so to 'armeabi-v7a' or 'x86'
but... i got this same error in two ways above.
```02-09 21:28:24.393 11663-11663/? E/linker: library "/system/lib/libopencv_core.so" ("/system/lib/libopencv_core.so") needed or dlopened by "/system/lib/libnativeloader.so" is not accessible for the namespace: [name="classloader-namespace", ld_library_paths="", default_library_paths="/data/app/com.sdjzu.wangfuying.javacv-1/lib/arm:/data/app/com.sdjzu.wangfuying.javacv-1/base.apk!/lib/armeabi-v7a", permitted_paths="/data:/mnt/expand:/data/data/com.sdjzu.wangfuying.javacv"]
02-09 21:28:24.400 11663-11663/? E/linker: library "/system/lib/libopencv_imgproc.so" ("/system/lib/libopencv_imgproc.so") needed or dlopened by "/system/lib/libnativeloader.so" is not accessible for the namespace: [name="classloader-namespace", ld_library_paths="", default_library_paths="/data/app/com.sdjzu.wangfuying.javacv-1/lib/arm:/data/app/com.sdjzu.wangfuying.javacv-1/base.apk!/lib/armeabi-v7a", permitted_paths="/data:/mnt/expand:/data/data/com.sdjzu.wangfuying.javacv"]
```
is it a linker error ?

**Comment by 805447391** [comment]
but i find it works well 👍 
```
package com.sdjzu.wangfuying.javacv;

import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import org.bytedeco.javacpp.avcodec;
import org.bytedeco.javacpp.opencv_core;
import org.bytedeco.javacpp.opencv_core.Mat;
import org.bytedeco.javacpp.opencv_core.Scalar;
import org.bytedeco.javacv.AndroidFrameConverter;
import org.bytedeco.javacv.OpenCVFrameConverter;

import static org.bytedeco.javacpp.avcodec.AV_CODEC_ID_H264;
import static org.bytedeco.javacpp.avcodec.avcodec_find_decoder;
import static org.bytedeco.javacpp.avcodec.avcodec_register_all;
import static org.bytedeco.javacpp.opencv_core.CV_8UC4;
import static org.bytedeco.javacpp.opencv_imgproc.CV_INTER_CUBIC;
import static org.bytedeco.javacpp.opencv_imgproc.resize;

public class MainActivity extends AppCompatActivity {

    private final String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // JavaCV
        Mat mat = new Mat(500, 500, CV_8UC4, new Scalar(100, 100, 100, 100));
        Log.e(TAG, "Mat's elemSize = " + mat.elemSize());
        Mat mat2 = new Mat();
        resize(mat, mat2, new opencv_core.Size(10, 10), 0, 0, CV_INTER_CUBIC);

        // FFmpeg
        avcodec_register_all();
        avcodec.AVCodec avCodec = avcodec_find_decoder(AV_CODEC_ID_H264);
        Log.e("TAG", "avCodec = " + avCodec.id());

        // Android native bitmap
        Bitmap bitmap = new AndroidFrameConverter().convert(new OpenCVFrameConverter.ToMat().convert(mat));
        getWindow().getDecorView().setBackground(new BitmapDrawable(bitmap));

        // release
        mat2.release();
        mat.release();

    }
}

```

**Comment by saudet** [comment]
Probably the same as issue #904, but happy to hear that it works anyway. :)

**Comment by 805447391** [comment]
@saudet thanks very much, i will try another way in #904, and close this issue later.have a nice day ! 💯

**Comment by 805447391** [comment]
here is my new gradle configuration:
```
apply plugin: 'com.android.application'

android {
    compileSdkVersion 26
    defaultConfig {
        applicationId "com.sdjzu.wangfuying.javacv"
        minSdkVersion 19
        targetSdkVersion 26
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
    splits {
        abi {
            enable true
            reset()
            include 'x86', 'armeabi', 'armeabi-v7a'
            universalApk true
        }
    }
}

dependencies {
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    implementation 'com.android.support:appcompat-v7:26.1.0'
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.1'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
    // cv
    compile group: 'org.bytedeco', name: 'javacv', version: '1.4'
    compile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.4.0-1.4', classifier: 'android-arm'
    compile group: 'org.bytedeco.javacpp-presets', name: 'opencv', version: '3.4.0-1.4', classifier: 'android-x86'
    compile group: 'org.bytedeco.javacpp-presets', name: 'ffmpeg', version: '3.4.1-1.4', classifier: 'android-arm'
    compile group: 'org.bytedeco.javacpp-presets', name: 'ffmpeg', version: '3.4.1-1.4', classifier: 'android-x86'
}

```
but it always display that error (same as issue title),and works well.
so i decide ti ignore this in mind ;)
thanks all contributors!

**Comment by saudet** [comment]
Yes, I think that's an issue with some devices, so let's keep this issue open at least until we figure out a work around... Thanks!

**Comment by saudet** [comment]
This should now be fixed with commit https://github.com/bytedeco/javacpp/commit/f1f21ae38aba00972992aa03b77fed763377ba77 which has been included in version 1.4.2. Thanks for reporting and let me know if you are still having issues with that!
